src/ledger_write_stage.rs

```diff
@@ -0,0 +1,89 @@
+//! The `ledger_write_stage` module implements the ledger write stage. It
+//! writes entries to the given writer, which is typically a file or
+//! stdout
+
+use counter::Counter;
+use entry::{EntryReceiver, EntrySender};
+use ledger::LedgerWriter;
+use log::Level;
+use result::{Error, Result};
+use service::Service;
+use std::sync::atomic::AtomicUsize;
+use std::sync::mpsc::RecvTimeoutError;
+use std::thread::{self, Builder, JoinHandle};
+use std::time::Duration;
+
+pub struct LedgerWriteStage {
+    write_thread: JoinHandle<()>,
+}
+
+impl LedgerWriteStage {
+    pub fn write(
+        ledger_writer: Option<&mut LedgerWriter>,
+        entry_receiver: &EntryReceiver,
+        forwarder: &Option<EntrySender>,
+    ) -> Result<()> {
+        let mut ventries = Vec::new();
+        let mut received_entries = entry_receiver.recv_timeout(Duration::new(1, 0))?;
+
+        loop {
+            ventries.push(received_entries);
+
+            if let Ok(n) = entry_receiver.try_recv() {
+                received_entries = n;
+            } else {
+                break;
+            }
+        }
+
+        if let Some(ledger_writer) = ledger_writer {
+            for entries in ventries {
+                for e in &entries {
+                    ledger_writer.write_entry_noflush(&e)?;
```

Comment:
I really don't want to share details of flush() with ledger_writer clients, just remove flush() from ledger